openxla-nvgpu/compiler/src/openxla/compiler/nvgpu/Transforms/ConvertMHLOToCUDNN.cpp

```diff
@@ -0,0 +1,97 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <numeric>
+#include <algorithm>
+
+#include "mhlo/IR/hlo_ops.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/IR/Builders.h"
+#include "mlir/IR/BuiltinAttributes.h"
+#include "mlir/IR/BuiltinTypes.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+#include "openxla/compiler/nvgpu/Dialect/CUDNN/IR/CUDNNDialect.h"
+#include "openxla/compiler/nvgpu/Dialect/CUDNN/IR/CUDNNOps.h"
+#include "openxla/compiler/nvgpu/Transforms/Passes.h"
+
+#define GEN_PASS_DEF_CONVERTMHLOTOCUDNN
+#include "openxla/compiler/nvgpu/Transforms/Passes.h.inc"
+
+using namespace mlir;
+
+namespace openxla::compiler::nvgpu {
+
+static std::optional<APFloat> getFloatSplatValue(Value value) {
+  auto op = value.getDefiningOp<mhlo::ConstantOp>();
```

Comment:
I have a feeling that `mlir/IR/Matchers.h` can make it simpler, not sure about splat thing though

Reply:
Done.